### ISSUE TITLE
Question: Own Dataset

### ISSUE THREAD

**Issue by Ademord** [chensong1995/HybridPose#28]
Hello
I was wondering if you could provide a bit more documentation on how to train on own dataset. Say I want to use this project for getting the 6D pose of chairs. Could you clarify:

- what format does my dataset need?
- what about the labeling?
- where to place what in the folder structure?
- finally, how do we run the training and then evaluate results (commands to execute)?

Loved your project work. I'm a huge fan and I am excited to learn from your steps!

**Comment by chensong1995** [comment]
Hello Ademord,

Thanks for trying out our code!

Our data format is actually pretty simple. I suggest you take a look at either  [linemod.py](https://github.com/chensong1995/HybridPose/blob/master/lib/datasets/linemod.py) or [occlusion_linemod.py](https://github.com/chensong1995/HybridPose/blob/master/lib/datasets/occlusion_linemod.py).

To support a new custom dataset, you will need to create a `torch.utils.data.Dataset` class, whose `__getitem__()` returns a dictionary with [these keys](https://github.com/chensong1995/HybridPose/blob/master/lib/datasets/occlusion_linemod.py#L158). You then need to use this `Dataset` class in [train_core.py](https://github.com/chensong1995/HybridPose/blob/master/src/train_core.py). You should also make a choice on the covariance matrix [here](https://github.com/chensong1995/HybridPose/blob/master/trainers/coretrainer.py#L419). The two choices have more-or-less the same performance. You can try both of them.

I use the Farthest Point Sampling algorithm to label 3D keypoints, and [SymSeg](https://github.com/aecins/symseg) to label the 3D symmetry plane. Please refer to #13 for projecting the 3D intermediate representations to 2D using ground-truth poses.

You can put your dataset anywhere on the disk as long as it is recognizable by your custom `torch.utils.data.Dataset` instance.

You use `LD_LIBRARY_PATH=lib/regressor:$LD_LIBRARY_PATH python src/train_core.py` to train the prediction network and obtain predictions. You use `python src/evaluate.py` to evaluate the performance.

I hope this helps!
